### DISCUSSION TITLE
Can I add a parent to theia's container?

### DISCUSSION THREAD

**Discussion by TotooriaHyperion** [eclipse-theia/theia#11448] [Q&A] [answered]
I'm using theia as a micro-frontend module, where I have some global services in my host app, which may be used by some plugin which is plugged to theia. These services is also used by non-theia modules such as other micro-frontend async module.
The problem is, I want to inject global services in to plugins as well as theia provided services. And my plugin module can't have both theia's container and the global container as parentContainer at the same time.
Is there anyway to give a parentContainer to theia's container(which is inside the generated code frontend.js)?

Thanks.

**Comment by msujew** [comment, marked as answer]
Hi @TotooriaHyperion,

I don't think the design of the generated code allows to easily add a parent to Theia's DI container. You could use something as [patch-package](https://www.npmjs.com/package/patch-package), to modify the generated code in a build step, but I wouldn't recommend that.

Instead, you could provide a common `ContainerModule` that you then load into both DI containers. Would that work for you?

**Reply by TotooriaHyperion** [reply]
Yes, good idea. I can have a isolate rootContainer and provide a ContainerModule to bind all the services through toDynamicValue refer to rootContainer.

```typescript
export const rootContainer = new Container({ defaultScope: 'Singleton' });
rootContainer.bind(RouterService).toSelf();

export const RootServiceModule = new ContainerModule((bind) => {
  bind(RouterService).toDynamicValue(() => rootContainer.get(RouterService));
});
```
